### PR TITLE
[Lian Jiade] iP

### PR DESCRIPTION
**Pull request by CSjiade**

### CODE REVIEW

File: src/main/java/Deadline.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    super(description);
+    this.by = dateTime(by);
+  }
+
+  public Deadline(String description, LocalDateTime by) {
+    super(description);
+    this.by = by;
+  }
+
+  private LocalDateTime dateTime(String by) {
```
Comment (rajobasu) [inline src/main/java/Deadline.java:20]: Usually I think it’s better to use method names that are verbs rather!

File: src/main/java/Deadline.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      format = DateTimeFormatter.ofPattern("d/MM/yyyy Hmm");
+      date = LocalDateTime.parse(by, format);
+    } catch (DateTimeParseException e) {
+      System.out.println("Please use d/MM/yyyy Hmm for your date");
+      dateError = true;
+    }
+    return date;
+  }
+
+  private boolean getdateError() {
```
Comment (rajobasu) [inline src/main/java/Deadline.java:33]: Just a small typo I guess, getDateError()?

File: src/main/java/Duke.java [hunk 3]
```diff
@@ -1,10 +1,73 @@
...
+   * Outputs the greeting and initiaties the userinput parase command to take in inputs from usher.
+   */
+
+  public void run() throws IOException, DescriptionError {
+    ui.greeting();
+    //Parser parse = new Parser();
+    //parse.userinput(tasks, storage);
+  }
+
+  public void listtask() {
```
Comment (rajobasu) [inline src/main/java/Duke.java]: Again I guess should be listTask()?

File: src/main/java/Parser.java [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    this.ui = new Ui();
+    store = storage;
+    tasklist = tasks;
+  }
+
+  /**
+   * Reads in the input by the user from the commandline and processes it.
+   */
+
+  public String userinput(String command) throws IOException, DescriptionError {
```
Comment (rajobasu) [inline src/main/java/Parser.java:23]: This should also be userInput()

File: src/main/java/TaskList.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    String output = "\tNoted. I've removed this task: \n" + task + "\n" +
+            "Now you have " + this.tasklist.size() + " tasks in the list.";
+    return output;
+  }
+
+  /**
+   * Outputs matching taks in the arraylist to a keyword
+   */
+
+  public String findtask(String keyword) {
```
Comment (rajobasu) [inline src/main/java/TaskList.java]: This should also be findTask

File: src/test/java/DeadLinetest.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class DeadLinetest {
+
+  @Test
+  public void deadLinetest() {
```
Comment (jrvslam) [inline src/test/java/DeadLinetest.java:8]: should be deadlineTest()?

File: src/main/java/Deadline.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  @Override
+  public String saveText() {
+    return "D" + "," + this.getStatusIcon() + "," + this.getDescription() + "," + this.by;
+  }
+
+
+
+}
```
Comment (jrvslam) [inline src/main/java/Deadline.java:63]: Small thing, but should the tab spacings be 4 space wide?

File: src/main/java/Duke.java [hunk 8]
```diff
@@ -1,10 +1,73 @@
...
+  public Duke() throws IOException {
+    ui = new Ui();
+    storage = new Storage("duke.txt");
+    try {
+      tasks = new TaskList(storage.load());
+      System.out.println(tasks.getTasklist());
+    } catch (FileNotFoundException e) {
+      System.out.println("No existing Storage File");
     }
+  }
```
Comment (jrvslam) [inline src/main/java/Duke.java]: Similarlly, should each tab spacing be 4 spaces wide?

File: src/main/java/DukeException.java [hunk 9]
```diff
@@ -0,0 +1,9 @@
+public class DukeException extends Exception {
+
+  protected String msg;
+
+  public DukeException(String msg) {
+    super(msg);
+  }
+
+}
```
Comment (jrvslam) [inline src/main/java/DukeException.java:9]: Once again, should the tab spacings be 4 spaces wide?

File: src/main/java/Event.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  @Override
+  public String toString() {
+    return "[E]" + super.toString() + " (at:" + by + ")";
+  }
+
+  @Override
+  public String saveText() {
+    return "E" + "," + this.getStatusIcon() + "," + this.getDescription() + "," + this.by;
+  }
+}
```
Comment (jrvslam) [inline src/main/java/Event.java:28]: Here again too, should the tab spacings be 4 spaces wide?

File: src/main/java/Parser.java [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+      } else {
+        try {
+          throw new UnknownInputError("☹ OOPS!!! I'm sorry, "
+                     + "but I don't know what that means :-(");
+        } catch (UnknownInputError e) {
+          outputMessage += e.getMessage();
+        }
+      }
+      return outputMessage;
+    }
```
Comment (jrvslam) [inline src/main/java/Parser.java]: Here too, should the tab spacings be 4 spaces wide?

File: src/main/java/Storage.java [hunk 12]
```diff
@@ -0,0 +1,98 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
```
Comment (jrvslam) [inline src/main/java/Storage.java:10]: For this class too, should the tab spacings be 4 spaces wide?

File: src/main/java/Task.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
+public class Task {
```
Comment (jrvslam) [inline src/main/java/Task.java:1]: Similarly here too, should the spacings be 4 spaces wide?

**Review by jrvslam — COMMENTED** [review]
Overall the code is very neat! Great use of classes and oop! I noted some minor layout errors but other than that it's all good! Good job! :)

File: src/main/java/Duke.java [hunk 14]
```diff
@@ -1,22 +1,36 @@
 import java.io.FileNotFoundException;
```
Comment (Andrewzhang217) [inline src/main/java/Duke.java:1]: Generally the code is very clean, good job!

File: src/main/java/Duke.java [hunk 15]
```diff
@@ -25,15 +39,18 @@ public Duke(String filePath) throws IOException {
...
   public void run() throws IOException, DescriptionError {
-    ui = new Ui();
     ui.greeting();
-    Parser parse = new Parser();
-    parse.userinput(tasks, storage);
+    //Parser parse = new Parser();
+    //parse.userinput(tasks, storage);
   }
 
   public void listtask() {
```
Comment (Andrewzhang217) [inline src/main/java/Duke.java]: Should you use Camel case to name this method?

File: src/main/java/MainWindow.java [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Background;
+import javafx.scene.layout.BackgroundFill;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+
```
Comment (Andrewzhang217) [inline src/main/java/MainWindow.java:9]: Should you delete the space between two import statements?

**Review by Andrewzhang217 — COMMENTED** [review]
Generally very clean; well done! Only need to change a few method names using camelcase.

**Pull request closed without merging**